modules/system/classes/MailManager.php

```diff
@@ -93,21 +96,73 @@ public function addContentToMailer($message, $code, $data, $plainOnly = false)
             return false;
         }
 
-        $this->addContentToMailerInternal($message, $template, $data, $plainOnly);
+        $html = $text = $template;
+
+        if ($plainOnly) {
+            $html = null;
+        }
+        $this->addContentToMailerInternal($message, $html, $text, $data);
 
         return true;
     }
 
     /**
-     * Internal method used to share logic between `addRawContentToMailer` and `addContentToMailer`
+     * Restore proper behavior in-line with Laravel Mailer
+     * Replaces both `addContentToMailer` and `addRawContentToMailer`
      *
      * @param \Illuminate\Mail\Message $message
-     * @param string $template
+     * @param string $view
+     * @param string $plain
+     * @param string $raw
      * @param array $data
-     * @param bool $plainOnly Add only plain text content to the message
      * @return void
      */
-    protected function addContentToMailerInternal($message, $template, $data, $plainOnly = false)
+    public function addContent($message, $view, $plain, $raw, $data)
+    {
+        $html = $text = null;
+
+        if (isset($view)) {
+            if (isset($this->templateCache[$view])) {
+                $html = $this->templateCache[$view];
+            } else {
+                $this->templateCache[$view] = $html = MailTemplate::findOrMakeTemplate($view);
+            }
+        }
+
+        if (isset($plain)) {
+            if (isset($this->templateCache[$plain])) {
+                $text = $this->templateCache[$plain];
+            } else {
+                $this->templateCache[$plain] = $text = MailTemplate::findOrMakeTemplate($plain);
+            }
+        }
+
+        // raw content will overwrite plain view content, as done in laravel
+        if (isset($raw)) {
+            $text = new MailTemplate;
+            $text->fillFromContent($raw);
+        }
+
+        // bailout if we have no content
+        if (!($html || $text)) {
```

Comment:
@mjauvin Need to do a proper null check here
```suggestion
        if (!isset($html) && !isset($text)) {
```